### PR TITLE
Database/mgmt fixes

### PR DESCRIPTION
**Pull request by Whatsuup**
# Pull Request for initial Implementation
## Backend
Make sure, you have done de following before merging this pull request:
- [x] Your code is **documented** using doc comments.
- [x] You added appropriate **logging** in your code.
- [x] You added **unit tests** to all your implemented functions.
- [x] You ran `cargo format` to format your code.
- [x] You ran `cargo clippy` to check for improvements.
- [x] You added the **class descriptions** of the classes you implemented in the **latex document** on overleaf.
      Make sure to mark any changes you made to the design documents specification.

### CODE REVIEW

**Comment by Whatsuup** [comment]
MealPlanManagement Tests work now. Interface and all relations are fixed.

NTBD:
- Add changes to overleaf
- Improve Mock
- Improve Docs
- Clippy in new code

File: backend/src/interface/persistent_data.rs [hunk 1]
```diff
@@ -26,77 +26,89 @@ pub enum DataError {
 }
```
Comment (Whatsuup) [inline backend/src/interface/persistent_data.rs:26]: Please check if these params are enough to perform each method on the database!

File: backend/src/interface/persistent_data.rs [hunk 2]
```diff
@@ -26,77 +26,91 @@ pub enum DataError {
...
-    /// Adds a new line entity to the database. Returns the new entity.
-    async fn insert_line(&self, name: &str) -> Result<Line>;
-    /// Adds a new meal entity to the database. Returns the new entity.
+    /// Adds a new canteen entity to the database.
+    /// Returns uuid of the new canteen.
+    async fn insert_canteen(&self, name: &str) -> Result<Uuid>;
+    /// Adds a new line entity to the database.
+    /// Returns uuid of the new line.
+    async fn insert_line(&self, name: &str) -> Result<Uuid>;
+    /// Adds a new meal entity to the database and creates a meal plan entry for the given day.
```
Comment (Whatsuup) [inline backend/src/interface/persistent_data.rs:90]: #[allow(clippy::too_many_arguments)] allowed in this case?

File: backend/src/interface/persistent_data.rs [hunk 3]
```diff
@@ -26,77 +26,91 @@ pub enum DataError {
...
+    /// Updates an existing canteen entity in the database.
+    /// Returns the canteen uuid.
+    async fn update_canteen(&self, uuid: Uuid, name: &str) -> Result<Uuid>;
+    /// Updates an existing line entity in the database.
+    /// Returns the line uuid.
+    async fn update_line(&self, uuid: Uuid, name: &str) -> Result<Uuid>;
+    /// Updates an existing meal entity in the database.
+    async fn update_meal(&self, uuid: Uuid, name: &str) -> Result<()>;
+    /// Adds a meal into the meal plan with the given params.
+    async fn add_meal_to_plan(
```
Comment (worldofjoni) [inline backend/src/interface/persistent_data.rs:66]: to add a meal to the mealplan the following (and nothing more) is neccesary:
- meal_id (food_id)
- line_id
- date
- price

Comment (worldofjoni) [reply backend/src/interface/persistent_data.rs:66]: so exchange canteen for line

File: backend/src/interface/persistent_data.rs [hunk 4]
```diff
@@ -26,77 +26,91 @@ pub enum DataError {
...
+    /// Adds a new line entity to the database.
+    /// Returns uuid of the new line.
+    async fn insert_line(&self, name: &str) -> Result<Uuid>;
+    /// Adds a new meal entity to the database and creates a meal plan entry for the given day.
+    #[allow(clippy::too_many_arguments)]
     async fn insert_meal(
         &self,
         name: &str,
         meal_type: MealType,
         price: Price,
```
Comment (worldofjoni) [inline backend/src/interface/persistent_data.rs:96]: price not needed here (only at meal plan)

File: backend/src/interface/persistent_data.rs [hunk 5]
```diff
@@ -26,77 +26,91 @@ pub enum DataError {
...
-        next_served: Date,
         allergens: &[Allergen],
         additives: &[Additive],
-    ) -> Result<Meal>;
-    /// Adds a new side entity to the database. Returns the new entity.
+        canteen_id: Uuid,
+        next_served: Date,
```
Comment (worldofjoni) [inline backend/src/interface/persistent_data.rs:100]: canteen and serve dat not needed here (infered from mealplan entry)

File: backend/src/interface/persistent_data.rs (hunk 2 above)
Comment (worldofjoni) [reply backend/src/interface/persistent_data.rs:90]: not neccesary

**Review by worldofjoni — CHANGES_REQUESTED** [review]
see comments

**Comment by Whatsuup** [comment]
this branch is now merge ready

**Review by worldofjoni — APPROVED** [review]

**Pull request merged**